UnityProject/Assets/Scripts/PlayGroups/PlayerMove.cs

```diff
@@ -250,32 +229,81 @@ private Vector3Int AdjustDirection(Vector3Int currentPosition, Vector3Int direct
                         //}
                         //Hit Pull obj
                         pna.CmdStopPulling(playerSync.PullingObject);
-
-                        return Vector3Int.zero;
+                        Interact(currentPosition, direction);
+                        return false;
                     }
                 }
             }
             if (!curMatrix.ContainsAt(newPos, gameObject) && curMatrix.IsPassableAt(currentPosition, newPos) && !isReplay)
             {
-                return direction;
+                return true;
             }
             //This is only for replay (to ignore any interactions with the pulled obj):
             if (playerSync.PullingObject != null)
             {
                 if (curMatrix.ContainsAt(newPos, playerSync.PullingObject))
                 {
-                    return direction;
+                    return true;
                 }
             }
 
             if (isReplay)
             {
-                return direction;
+                bool is_not_replay = false;
```

Comment:
What's the point of introducing this variable here? And you're passing that isNotReplay to the method that accepts isReplay, that makes no sense and brings a lot of confusion

Reply:
![Image](https://cdn.discordapp.com/attachments/312454684021620736/463715744727826433/GIF3321231.gif "Image") 
If it doesn't check it can move their, it ends up giving false locations to move to  Since if "isReplay" Returns always true